git-active-branches

```diff
@@ -1,9 +1,40 @@
 #!/bin/sh
 set -eu
 
-# TODO: Make the max age configurable
+since='3.weeks.ago'
+
+usage() {
+    echo "usage: git active-branches [--since|--after <date>]" >&2
+    echo >&2
+    echo "Unless specified, <date> defaults to \"$since\". For examples see:" >&2
+    echo "https://git-scm.com/book/en/v2/Git-Basics-Viewing-the-Commit-History" >&2
+    exit 2
+}
+
+if [ $# -gt 0 ]; then
+    case $1 in
+        # 'git log' accepts either '--since' or '--after' as synonyms
+        --since|--after)
+            [ $# -eq 2 ] || usage
+            since=$2
+            ;;
+        --since=*)
+            since=$(echo "$1" | sed 's/--since=\(.*\)/\1/')
+            ;;
+        --after=*)
+            since=$(echo "$1" | sed 's/--after=\(.*\)/\1/')
+            ;;
+        *)
+            usage
+            ;;
+    esac
+fi
 
 ( git local-branches; git remote-branches ) | while read branch; do
-    maybebranch=$(git log -1 --since=3.weeks.ago -s $branch)
+    maybebranch=$(
+        # set TRACE=1 in environment to show 'git log' commands as they're run
+        [ "${TRACE:-}" = 1 ] && set -x
```

Comment:
Adding support for `TRACE=1` isn't necessary. Instead of setting `TRACE=1` in the environment, you can already just `set -x` yourself and then invoke this script. That way, you can use the standard way to enable tracing that works with all scripts, not just this one.

Reply:
OK, thanks for being firm about that. I was going to see what you said and then probably surreptitiously sneak it into other places if you hadn't pushed back. I will remove the `TRACE` part entirely. `:)`

I'm not fixated at all on whether my commits end up in the source tree with _my_ name on them, but I would like to know whether you intend to squash commits (so intermediate commits like "oops, remove that" won't matter in the end), if I should start over from "master," or if the intermediate "oops" commits don't bother you.

I am learning, too, so I will defer to your judgment and make accomodations whatever you decide.

Reply:
Scripts run in a subshell, so they don't inherit the parent shell's `set -x`; that's why I use the TRACE thing the way I do. But it's easy enough to edit the script itself and just add the `x` to `set -eu`, when necessary for debugging.